### ISSUE TITLE
The indentation rule does not appear to work for switch / case ...

### ISSUE THREAD

**Issue by livelongai** [eslint/eslint#15360]
### Environment

Node version:  14.8.0
npm version: 8.1.3
Local ESLint version: 8.2.0
Global ESLint version: 8.2.0
Operating System: Darwin 20.6.0

I would assume that `case` should be indented relative to its `switch`.  However, eslint is marking this as an error and asking me to line `case` up with `switch`.  See pic below.

<img width="423" alt="Screen Shot 2021-11-26 at 6 09 11 PM" src="https://user-images.githubusercontent.com/94505039/143662148-fec071e4-f24c-4f71-825f-44e68f9dcdce.png">

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```

{
  "parserOptions": {
    "parser": "babel-eslint",
    "ecmaVersion": 12,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "rules": {
    "max-depth": ["error", 5],
    "semi": "error",
    "indent": ["error", 2],
    "eqeqeq": ["error", "always"],
    "template-curly-spacing": ["error", "always"]
  }
}


```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

2 space indenting no to show an error

### What actually happened?

2 space indenting showed an error

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The attachment from above is sublime text which is setup to use ESLint.  It works well as the rules above are enforced; however I found it odd on the way it wanted me to indent switch/case statements.
